### ISSUE TITLE
xrootd proxy mode door IPV6 issue

### ISSUE THREAD

**Issue by cfgamboa** [dCache/dcache#7022]
Hello,

From LXPLUS to BNL

Test 1. IPV4 works
```
[cgamboa@lxplus707 ~]$ xrdcp -f  -DSNetworkStack IPv4 root://dcqosdoor02.usatlas.bnl.gov:1097/pnfs/usatlas.bnl.gov/diskdata/new_test_hsi_20.21-localpush2 /tmp/test.1
security protocol 'ztn' disallowed for non-TLS connections.
[63.37kB/63.37kB][100%][==================================================][31.68kB/s]  
```

Test 2. IPV6 fails 
```
[cgamboa@lxplus707 ~]$ xrdcp -f  -DSNetworkStack IPv6 root://dcqosdoor02.usatlas.bnl.gov:1097/pnfs/usatlas.bnl.gov/diskdata/new_test_hsi_20.21-localpush2 /tmp/test.1
security protocol 'ztn' disallowed for non-TLS connections.
[0B/0B][100%][==================================================][0B/s]  
Run: [ERROR] Invalid redirect URL:  (source)
```

**The door proxying the request configuration** 

```
[xrootd-${host.name}Domain]
dcache.java.memory.heap=4096m
dcache.java.memory.direct=4096m
[xrootd-${host.name}Domain/xrootd]
xrootd.cell.name=Xrootd-${host.name}
xrootd.net.port=1097
xrootd.security.tls.mode = OFF
xrootd.net.proxy-transfers=true
xrootd.net.internal=10.42.64.38
```

Every time the test 2 is execute the test fails and there are sockets left opened.

```
netstat -lntp|grep 39082
tcp6       0      0 2620:0:210:1::d:21694   :::*                    LISTEN      39082/java          
tcp6       0      0 2620:0:210:1::d:22207   :::*                    LISTEN      39082/java          
tcp6       0      0 2620:0:210:1::d:23746   :::*                    LISTEN      39082/java          
tcp6       0      0 2620:0:210:1::d:20899   :::*                    LISTEN      39082/java          
tcp6       0      0 :::1097                 :::*                    LISTEN      39082/java          
tcp6       0      0 2620:0:210:1::d:20084   :::*                    LISTEN      39082/java        
```

The IPV6 external interface is `2620:0:210:1::d`

I can provide the entire client xrdcp -d 3 stack as needed

**Comment by alrossi** [comment]
Please attach the client log.   But it would be great to see (if there is any) more info from the door on why the redirect URL is invalid.   The fact that the sockets are left open suggests to me that there is an error not being caught properly by the door.
It would be even better if you could turn on debugging in the proxy door and include that output.   Thanks!

**Comment by cfgamboa** [comment]
Hello Al.

I have created a dCache RT www.dcache.org #10432 to upload the information requested

All the best,
Carlos

**Comment by alrossi** [comment]
Thanks Carlos.  I will close the GH issue and investigate.  Al


________________________________________________
Albert L. Rossi
Senior Software Developer
Scientific Computing Division, Scientific Data Services, Distributed Data Development
WH 566
Fermi National Accelerator Laboratory
Batavia, IL 60510
(630) 840-3023

________________________________

**Comment by cfgamboa** [comment]
Patch included on 8.2.15, https://github.com/dcache/dcache/commit/bea306c31093a8328ac874229ede2619d0cc280f
